src/command/lsp/mod.rs

```diff
@@ -0,0 +1,230 @@
+                        .ok()?
+                        .join(path),
+                )
+            } else {
+                Some(path.clone())
+            }
+        })
+        .ok_or_else(|| anyhow!("Could not find supergraph.yaml file."))?;
+    let supergraph_content_root = supergraph_yaml_path.parent().unwrap().to_path_buf();
+
+    let studio_client = client_config.get_authenticated_client(&lsp_opts.plugin_opts.profile)?;
+
+    // Get the SupergraphConfig in a form we can use
+    let supergraph_config = SupergraphConfigResolver::default()
+        .load_remote_subgraphs(&studio_client, None)
+        .await?
+        .load_from_file_descriptor(
+            &mut stdin(),
+            Some(&FileDescriptorType::File(supergraph_yaml_path.clone())),
+        )?;
+    let lazily_resolved_supergraph_config = supergraph_config
+        .lazily_resolve_subgraphs(&supergraph_content_root)
+        .await?;
+
+    // Build the service to spin up the language server
+    let (service, socket, _receiver) = ApolloLanguageServer::build_service(
+        Config {
+            root_uri: supergraph_content_root.to_string(),
+            enable_auto_composition: false,
+            force_federation: false,
+            disable_telemetry: false,
+        },
+        HashMap::from_iter(
+            lazily_resolved_supergraph_config
+                .subgraphs()
+                .iter()
+                .map(|(a, b)| (a.to_string(), b.schema().clone())),
+        ),
+    );
+
+    let language_server = service.inner().clone();
+
+    // Spawn a separate thread to handle composition and passing that data to the language server
+    tokio::spawn(async move {
+        // Create a supergraph binary
+        // TODO: Check defaulting behaviour here and see if we need to centralise
+        let federation_version = lazily_resolved_supergraph_config
+            .federation_version()
+            .clone()
+            .unwrap_or(FederationVersion::LatestFedTwo);
+
+        // TODO: Let the supergraph binary exist inside its own task that can respond to being re-installed etc.
+        let supergraph_binary =
+            InstallSupergraph::new(federation_version.clone(), client_config.clone())
+                .install(
+                    None,
+                    lsp_opts.plugin_opts.elv2_license_accepter,
+                    lsp_opts.plugin_opts.skip_update,
+                )
+                .await?;
+
+        // Spin up Runner
+        let mut stream = Runner::default()
+            .setup_subgraph_watchers(
+                lazily_resolved_supergraph_config.subgraphs().clone(),
+                &lsp_opts.plugin_opts.profile,
+                &client_config,
+                500,
+            )
+            .setup_supergraph_config_watcher(lazily_resolved_supergraph_config.clone())
+            .setup_composition_watcher(
+                lazily_resolved_supergraph_config
+                    .fully_resolve_subgraphs(&client_config, &studio_client)
```

Comment:
This method is new because in the original version of the refactor you either had a FullyResolvedSupergraphConfig (so all the elements had been reduced to SDL sources), or a LazilyResolvedSupergraphConfig (which still had the capacity to spin up FileWatchers etc.). 

The issue in this instance is that you need both (or specifically you need a way to obtain `FullyResolvedSubgraphs`). So I decided that it was best to implement a method to convert a LazilyResolvedSupergraphConfig to a FullyResolvedSubgraphs. Because this transformation is only possible one way as you lose information once they're fully resolved.

Happy to argue this one out, because it is a bit weird that the Runner requires a LazilyResolvedSupergraphConfig at the top and then FullyResolved in later methods but didn't want to touch that code too much due to the ongoing `dev` refactor